File: src/Functional/Elixir/singleton.exs
defmodule SingletonRegistry do
  @name __MODULE__

  def start_link do
    Agent.start_link(fn -> 0 end, name: @name)
  end

  def instance, do: Process.whereis(@name)
  def increment, do: Agent.update(@name, &(&1 + 1))
  def count, do: Agent.get(@name, & &1)
end

{:ok, _pid} = SingletonRegistry.start_link()
first = SingletonRegistry.instance()
second = SingletonRegistry.instance()
SingletonRegistry.increment()

IO.puts("same=#{if first == second, do: "true", else: "false"}")
IO.puts("count=#{SingletonRegistry.count()}")

File: src/Scripting/RubyRB/singleton.rb
require 'singleton'

class Registry
  include Singleton

  attr_reader :count

  def initialize
    @count = 0
  end

  def increment
    @count += 1
  end
end

first = Registry.instance
second = Registry.instance
first.increment

puts "same=#{first.equal?(second)}"
puts "count=#{second.count}"
